extensions/arc/src/ui/dashboards/postgres/postgresOverviewPage.ts

```diff
@@ -0,0 +1,225 @@
+		content.addItem(this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({ value: loc.serverGroupNodes, CSSStyles: titleCSS }).component());
+		const nodesTable = this.modelView.modelBuilder.declarativeTable().withProperties<azdata.DeclarativeTableProperties>({
+			width: '100%',
+			columns: [
+				{
+					displayName: loc.name,
+					valueType: azdata.DeclarativeDataType.string,
+					isReadOnly: true,
+					width: '30%',
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: cssStyles.tableRow
+				},
+				{
+					displayName: loc.type,
+					valueType: azdata.DeclarativeDataType.string,
+					isReadOnly: true,
+					width: '25%',
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: cssStyles.tableRow
+				},
+				{
+					displayName: loc.fullyQualifiedDomain,
+					valueType: azdata.DeclarativeDataType.string,
+					isReadOnly: true,
+					width: '45%',
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: cssStyles.tableRow
+				}
+			],
+			data: []
+		}).component();
+
+		const nodes = this.databaseModel.numNodes();
+		for (let i = 0; i < nodes; i++) {
+			nodesTable.data.push([
+				`${this.databaseModel.name()}-${i}`,
+				i === 0 ? loc.coordinatorEndpoint : loc.worker,
+				i === 0 ? `${endpoint.ip}:${endpoint.port}` : `${this.databaseModel.name()}-${i}.${this.databaseModel.name()}-svc.${this.databaseModel.namespace()}.svc.cluster.local`]);
+		}
+
+		content.addItem(nodesTable, { CSSStyles: { 'margin-bottom': '20px' } });
+		return root;
+	}
+
+	protected get toolbarContainer(): azdata.ToolbarContainer {
+		// New database
+		const newDatabaseButton = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: loc.newDatabase,
+			iconPath: IconPathHelper.add
+		}).component();
+
+		newDatabaseButton.onDidClick(async () => {
+			const name = await vscode.window.showInputBox({ prompt: loc.databaseName });
+			if (name === undefined) { return; }
+			const db: DuskyObjectModelsDatabase = { name: name }; // TODO support other options (sharded, owner)
+			await this.databaseModel.createDatabase(db)
+				.then(db => vscode.window.showInformationMessage(loc.databaseCreated(db.name)))
+				.catch(error => vscode.window.showErrorMessage(loc.databaseCreationFailed(db.name, error)));
+		});
+
+		// Reset password
+		const resetPasswordButton = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: loc.resetPassword,
+			iconPath: IconPathHelper.edit
+		}).component();
+
+		resetPasswordButton.onDidClick(async () => {
+			const password = await vscode.window.showInputBox({ prompt: loc.newPassword, password: true });
+			if (password === undefined) { return; }
+			await this.databaseModel.update(s => {
+				s.arc = s.arc ?? new DuskyObjectModelsDatabaseServiceArcPayload();
+				s.arc.servicePassword = password;
+			})
+				.then(_ => vscode.window.showInformationMessage(loc.passwordReset(this.databaseModel.fullName())))
```

Comment:
Since these functions are async you can just await the calls each separately (and then wrap in try/catch to handle errors)

Reply:
Is the suggestion to not use .then/.catch and instead do:

```
try {
    await this.databaseModel.createDatabase(db);
    vscode.window.showInformationMessage(...);
} catch (error) {
    vscode.window.showErrorMessage(...);
}
```

Reply:
Yeah - slightly cleaner and easier to follow